### ISSUE TITLE
Read on command

### ISSUE THREAD

**Issue by Jopp-gh** [mkiol/dsnote#109]
First of all, this project is awesome!

Still, I really don't know what to do with a text box interface, to be honest. It looks for me like an emergency solution at best.

Truly useful would be to read cursor highlighted text (or text highlighted on right-click) from any application (if applications permit)
On Mac, you can highlight any text and have it read out aloud on command. Super useful, independent and without any additional procedures or interfaces where you've to copy paste text.

**Comment by mkiol** [comment]
Thanks for the remarks and ideas.

I'm not super sure that this app is for these kind of tasks. At least this is not a primary use case. I think there are many better tools from "accessibility" space that are far more suitable for your need. I must say I didn't check, but most likely Orca can read whatever is under cursor. You can integrate Orca with Piper voices, for instance via [Pied](https://pied.mikeasoft.com) app, and achieve something close to what you've described.

The "accessibility" features implemented in Speech Note are very limited right now. You can read what is in the clipboard using keyboard shortcut. Perhaps this is something closer to what you are looking for?  You can copy whatever you want to the clipboard and use it `Ctrl+Alt+Shift+E` (default global shortcut) to read it.

**Comment by Jopp-gh** [comment]
>I think there are many better tools from "accessibility" space that are far more suitable for your need. I must say I didn't check, but most likely Orca can read whatever is under cursor. You can integrate Orca with Piper voices, for instance via Pied app, and achieve something close to what you've described.

Agreed, Orca is better suited for the task. I don't know Pied and will check it out, thanks!

SpeechNote is great because of the good and easy to understand ui design and the large list of modern voices - simply wonderful!

Espeak and older voices are hard to follow and hard to understand. Modern voices are easy to understand,  even at higher speeds. 

For the text input we could use xsel - I just don't know how to talk to SpeechNote's api , because I downloaded SpeechNote as flatpak.

Ideally a command would look like
`xsel | speechnote -v VoiceName`

And to save spoken text:
`xsel | speechnote -v VoiceName -out ~/path/to/test.mp3`

> You can read what is in the clipboard using keyboard shortcut. Perhaps this is something closer to what you are looking for?

Indeed, an useful feature.

**Comment by mkiol** [comment]
> And to save spoken text:
xsel | speechnote -v VoiceName -out ~/path/to/test.mp3

Actually, these kind of use case is already directly supported in cli interface by [Piper](https://github.com/rhasspy/piper) or [Coqui](https://github.com/coqui-ai/TTS). 

Most likely Speech Note is too "fat" for command line usage. Especially I strongly encourage you to try Piper that is designed to be light and provides good quality voices.

**Comment by mkiol** [comment]
It seems the answer is satisfactory, so I'm closing.
